src/nspawn/nspawn-oci.c

```diff
@@ -819,7 +819,7 @@ static int oci_device_file_mode(const char *name, JsonVariant *v, JsonDispatchFl
                 return json_log(v, flags, SYNTHETIC_ERRNO(ERANGE),
                                 "fileMode out of range, refusing.");
 
-        *mode = m;
+        *mode |= m;
```

Comment:
Then `oci_device_type()` should be also updated?

Please add a brief comment here why the result is ORed. Otherwise, we may accidentally drop it again.

Reply:
D'oh, indeed!